keras_nlp/utils/text_generation.py

```diff
@@ -17,6 +17,59 @@
 import tensorflow as tf
 
 
+def _validate_prompt(prompt):
+    """
+    Validate the prompt and reformat for use.
+
+    Args:
+        prompt: a list or a Tensor, can be 1D or 2D, the initial tokens to
+            append generated tokens.
+
+    Returns:
+        a 2D Tensor, the prompt with shape [batch_size, max_length].
+    """
+    if isinstance(prompt, tf.RaggedTensor):
+        raise ValueError(
+            "RaggedTensor `prompt` is not supported, please "
+            "provide `prompt` as a list or Tensor."
+        )
+    if not isinstance(prompt, tf.Tensor):
+        prompt = tf.convert_to_tensor(prompt)
+    input_is_1d = prompt.shape.rank == 1
+    if input_is_1d:
+        prompt = prompt[tf.newaxis, :]
+    return prompt, input_is_1d
+
+
+def _mask_tokens_after_end_token(
+    prompt, max_length, end_token_id, pad_token_id
+):
+    """
+    Mask the tokens after the end token.
+
+    Args:
+        prompt: a 2D Tensor, the prompt with shape [batch_size, max_length].
+        max_length: an integer, the maximum length of the prompt.
+        end_token_id: an integer, the id of the end token.
+        pad_token_id: an integer, the id of the padding token.
+
+    Returns:
+        a 2D Tensor, the masked prompt with shape [batch_size, max_length]. All
+        tokens after encountering `end_token_id` will be replaced with
+        `pad_token_id`.
+    """
+    # Mask out tokens after `end_token_id` is encountered.
+    # Find index of first end_token_id.
+    end_indices = tf.math.argmax(prompt == end_token_id, -1)
+    # Use max_length if no `end_token_id` is found.
+    end_indices = tf.where(end_indices == 0, max_length, end_indices)
+    # Build a mask including end_token and replace tokens after end_token
+    # with `pad_token_id`.
+    valid_indices = tf.sequence_mask(end_indices + 1, maxlen=max_length)
+    prompt = tf.where(valid_indices, prompt, pad_token_id)
```

Comment:
return directly

Reply:
Does this mean I should do "return tf.where(valid_indices, prompt, pad_token_id)"? Just edited